lib/valueflow.cpp

```diff
@@ -5462,12 +5462,14 @@ static bool isInBounds(const ValueFlow::Value& value, MathLib::bigint x)
     return true;
 }
 
-static const ValueFlow::Value* getCompareIntValue(const std::list<ValueFlow::Value>& values, std::function<bool(MathLib::bigint, MathLib::bigint)> compare)
+static const ValueFlow::Value* getCompareIntValue(const std::list<ValueFlow::Value>& values, std::function<bool(MathLib::bigint, MathLib::bigint)> compare, bool ignoreMinusOne)
 {
     const ValueFlow::Value* result = nullptr;
     for (const ValueFlow::Value& value : values) {
         if (!value.isIntValue())
             continue;
+        if (ignoreMinusOne && value.intvalue == -1 && value.bound == ValueFlow::Value::Bound::Upper)
```

Comment:
Why do we ignore `-1`? If `-1` should be ignored then it shouldn't be set in the first place.